parser/parser.go

```diff
@@ -208,6 +208,134 @@ func parseUnary(p *Parser) (ast.Node, error) {
 	return ast.NewUnaryExpr(tok.Type, expr), nil
 }
 
+func parseIdentExpr(p *Parser) (ast.Node, error) {
+	tok := p.lookahead[0]
+	p.scry(1)
+	next := p.lookahead[1]
+
+	// eg.: console.
+	if next.Type == token.Dot {
+		p.forget(1)
+		return parseMemberExpr(p, ast.NewIdent(tok.Value))
+	}
+
+	// eg.: console(
+	if next.Type == token.LParen {
+		return parseCallExpr(p)
+	}
+
+	if next.Type != token.EOF {
+		return nil, p.errorf(next, "unexpected token '%s'", next)
+	}
+
+	p.forget(2)
+
+	return ast.NewIdent(tok.Value), nil
+}
+
+// state:
+// lookahead[0] = token.Dot
+func parseMemberExpr(p *Parser, object ast.Node) (ast.Node, error) {
+	p.forget(1)
+
+	tok := p.next()
+	if tok.Type != token.Ident {
+		return nil, p.errorf(tok, "unexpected %s", tok.Value)
+	}
+
+	member := ast.NewMemberExpr(object, ast.NewIdent(tok.Value))
+
+	// TODO(i4k): Discuss this!
+	// We can just return the MemberExpr here but then if next token is '('
+	// then the main parser loop will need to retrieve the last parsed node
+	// to use it as the object in the CallExpr.
+	// Going into the CallExpr parser from here avoid to keep some states
+	// in the parser structure but can lead to some code duplicates also.
+	// I'm not sure of what's the best approach yet... Coding the naive one
+	// for now.
+	p.scry(1)
+
+	tok = p.lookahead[0]
+	if tok.Type == token.LParen {
+		return parseMemberFuncall(p, member)
+	}
+
+	if tok.Type == token.Dot {
+		return parseMemberExpr(p, member)
+	}
+
+	if tok.Type != token.EOF {
+		panic("not expected")
```

Comment:
Got a feeling that this panic message could have a little more detail on the unexpected token =P